src/Helper/ObjectParser.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Helper;
+
+use JetBrains\PhpStorm\ArrayShape;
+use JetBrains\PhpStorm\ExpectedValues;
+use ReflectionAttribute;
+use ReflectionObject;
+use ReflectionProperty;
+use Yiisoft\Validator\AfterInitAttributeEventInterface;
+use Yiisoft\Validator\RuleInterface;
+
+use function array_key_exists;
+
+final class ObjectParser
+{
+    /**
+     * @psalm-var array<string, array<string, array>>
+     */
+    #[ArrayShape([
+        [
+            'rules' => 'array',
```

Comment:
It was an iterable before. How we can guarantee that it's always an array?

Reply:
[Here](https://github.com/yiisoft/validator/blob/1304c80c9f2343bde15cd3db7f6b732708c4ba54/src/Helper/ObjectParser.php#L69) rules always array.